code/modules/economy/worth_cash.dm

```diff
@@ -164,36 +164,147 @@
+		var/obj/item/charge_stick/sender = W
+		var/datum/extension/lockable/W_lock = get_extension(W, /datum/extension/lockable)
+		if(lock.locked)
+			to_chat(user, SPAN_NOTICE("Cannot transfer funds to a locked [src]."))
+			return TRUE
+		if(W_lock.locked)
+			to_chat(user, SPAN_NOTICE("Cannot transfer funds from a locked [W]."))
+			return TRUE
+		if(sender.currency != currency)
+			to_chat(user, SPAN_WARNING("Mismatched currency detected. Unable to transfer."))
+			return TRUE
+		var/amount = input(user, "How much of [sender.loaded_worth] do you want to transfer?", "[sender] transfer", "0") as null|num
+		if(!amount)
+			return TRUE
+		if(amount < 0 || amount > sender.loaded_worth)
+			to_chat(user, SPAN_WARNING("Enter a valid number between 1 and [sender.loaded_worth] to transfer."))
+			return TRUE
+		sender.loaded_worth -= amount
+		loaded_worth += amount
+		sender.update_icon()
+		update_icon()
+		var/decl/currency/cur = decls_repository.get_decl(currency)
+		to_chat(user, SPAN_NOTICE("Completed transfer of [amount] [cur.name]."))
+		return TRUE
+	
+	if(lock.attackby(W, user))
+		return TRUE
+	return ..()
+
+/obj/item/charge_stick/emag_act(var/remaining_charges, var/mob/user, var/feedback)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	.= lock.emag_act(remaining_charges, user, feedback)
+
+/obj/item/charge_stick/attack_self(var/mob/user)
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	lock.ui_interact(user)		
+
+/obj/item/charge_stick/proc/is_locked()
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	return lock.locked
+
+/obj/item/charge_stick/on_update_icon()
+	. = ..()
+	if(get_world_inventory_state() == ICON_STATE_WORLD)
+		icon_state = "world"
+		return 
+
+	icon_state = grade
+	var/datum/extension/lockable/lock = get_extension(src, /datum/extension/lockable)
+	if(lock.locked)
+		return
+
+	if(loaded_worth > 999999)
+		overlays += image(icon, "9__")
+		overlays += image(icon, "_9_")
+		overlays += image(icon, "__9")
+		return
+
+	var/h_thou = loaded_worth / 100000
+	var/t_thou = (loaded_worth - (Floor(h_thou) * 100000)) / 10000
+	var/thou = (loaded_worth - (Floor(h_thou) * 100000) - (Floor(t_thou) * 10000)) / 1000
+	overlays += image(icon, "[Floor(h_thou)]__")
+	overlays += image(icon, "_[Floor(t_thou)]_")
+	overlays += image(icon, "__[Floor(thou)]")
+
+/obj/item/charge_stick/copper
+	grade = "copper"
+	max_worth = 20000
+	lock_type = /datum/extension/lockable/charge_stick/copper
+
+/obj/item/charge_stick/silver
+	grade = "silver"
+	max_worth = 50000
+	lock_type = /datum/extension/lockable/charge_stick/silver
+
+/obj/item/charge_stick/gold
+	grade = "gold"
+	max_worth = 200000
+	lock_type = /datum/extension/lockable/charge_stick/gold
+
+/obj/item/charge_stick/platinum
+	grade = "platinum"
+	max_worth = 500000
+	lock_type = /datum/extension/lockable/charge_stick/platinum
+
+/atom/movable/proc/GetChargeStick()
```

Comment:
need to add 
```
/obj/item/charge_stick/GetChargeStick()
  return src
```
otherwise wouldn't be able to pay in vendomats with credsticks directly